### DISCUSSION TITLE
Is it possible to keep the folder note title?

### DISCUSSION THREAD

**Discussion by ooker777** [ObsidianPublisher/obsidian-github-publisher#241] [Q&A — Plugin] [answered]
if I have `a/a.md` and set the folder note config to rename it to index.md, is it possible to automatically add `title: a` to yaml metadata?

**Comment by Mara-Li** [comment]
Probably with a regex it can be possible. I prefer to use templater to do that. I have the habit to use the `title` key before using my own plugins so my workflows is pretty smooth.

**Comment by Mara-Li** [comment]
The problem to add an option to edit the frontmatter in the "server file" is that I need to completely parse the frontmatter (as it is done in the GitHub deleting) manually, add the key and rewrite the frontmatter. 
I don't know if I can use the new frontmatter API without editing the "obsidian file" and broke every parsing. 

For the moment, keep the file content with reading the cache and use the strings. If I need to edit the frontmatter, it musts be done in first and I dunno. 
I think it's possible but I need to ask and test before.

TBH it can be a good option, so I will finish with my actual work (my little chat GPT bot) and keep an idea on it.

**Comment by Mara-Li** [comment, marked as answer]
See #242 for followup (i close the discussion)
